File: inheritanceex.py

class Parent:
    result=200
    def m1(self):
        print("am from m1 method of Parent class")
    def m2(self):
        print("am from m2 method of Parent class")
class Child(Parent):
    def m3(self):
        print("am from m3 method of child class")
obj=Parent()
obj.m1()
obj.m2()
print(obj.result)
obj1=Child()
obj1.m3()
obj1.m1()
obj1.m2()
print(obj1.result)

File: InheritanceEx.java
package com.astra.demo;

import java.util.Date;

class SuperParent {//object
	public void m1() {
		System.out.println("am from m1 method");
	}

	public void m2() {
		System.out.println("am from m2 method");
	}
}

class ParentEx extends SuperParent {
	public void displayAge() {
		int age = 12;
		System.out.println("Age is  :" + age);
	}

	public String displayMessage() {
		return "Hello Good Afternoon";
	}
}

public class InheritanceEx extends SuperParent {

	public void displayDate() {
		Date date = new Date();
		System.out.println("Today's Date : " + date);
	}

	public static void main(String[] args) {
		InheritanceEx obj = new InheritanceEx();
		obj.displayDate();
//		System.out.println(obj.displayMessage());
//		obj.displayAge();
		obj.m1();
		obj.m2();
	
	}

}
